app/components/carousel.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState } from "react";
+import type { Photo } from "./dummy-data";
+
+interface CarouselProps {
+    images: Photo[];
+    roomType: string;
+    address: string;
+}
+
+export function Carousel(props) {
```

Comment:
Please use anonymous function declaration throughout instead of function keyword. This prevents functions from being hoisted. This isn't a breaking change, but it is best practice and can prevent unexpected behavior. 

Also, make sure you are using the type you defined, CarouselProps, rather than just "props" which implicitly has any type: 
```suggestion
export function Carousel(props) {
export const Carousel = ({images, roomType, address}: CarouselProps) => {
```